pkg/sdn/plugin/egress_network_policy.go

```diff
@@ -108,8 +108,11 @@ func (plugin *OsdnNode) syncEgressDNSPolicyRules() {
 			continue
 		}
 
-		plugin.egressPoliciesLock.Lock()
-		defer plugin.egressPoliciesLock.Unlock()
-		plugin.updateEgressNetworkPolicyRules(vnid)
+		func() {
```

Comment:
Used defer since it is safer than explicitly calling unlock after the functionality. 
If there is a crash/panic in updateEgressNetworkPolicyRules(), any explicit unlock after this method will not be called but defer ensures unlock is called.